packages/tx/src/baseTransaction.ts

```diff
@@ -377,21 +379,55 @@ export abstract class BaseTransaction<TransactionObject> {
     }
   }
 
-  protected _validateCannotExceedMaxInteger(values: { [key: string]: BN | undefined }, bits = 53) {
+  /**
+   * Validates that an object with BN values cannot exceed the specified bit limit.
+   * @param values Object containing string keys and BN values
+   * @param bits Number of bits to check (64 or 256)
+   * @param cannotEqual Pass true if the number also cannot equal one less the maximum value
+   */
+  protected _validateCannotExceedMaxInteger(
+    values: { [key: string]: BN | undefined },
+    bits = 256,
+    cannotEqual = false
+  ) {
     for (const [key, value] of Object.entries(values)) {
-      if (bits === 53) {
-        if (value?.gt(MAX_INTEGER)) {
-          const msg = this._errorMsg(`${key} cannot exceed MAX_INTEGER, given ${value}`)
-          throw new Error(msg)
-        }
-      } else if (bits === 256) {
-        if (value?.gte(TWO_POW256)) {
-          const msg = this._errorMsg(`${key} must be less than 2^256, given ${value}`)
+      switch (bits) {
+        case 64:
+          if (cannotEqual) {
+            if (value?.gte(MAX_UINT64)) {
+              const msg = this._errorMsg(
+                `${key} cannot equal or exceed MAX_UINT64 (2^64-1), given ${value}`
+              )
+              throw new Error(msg)
+            }
+          } else {
+            if (value?.gt(MAX_UINT64)) {
+              const msg = this._errorMsg(`${key} cannot exceed MAX_UINT64 (2^64-1), given ${value}`)
+              throw new Error(msg)
+            }
+          }
+          break
+        case 256:
+          if (cannotEqual) {
+            if (value?.gte(MAX_INTEGER)) {
+              const msg = this._errorMsg(
+                `${key} cannot equal or exceed MAX_INTEGER (2^256-1), given ${value}`
+              )
+              throw new Error(msg)
```

Comment:
As mentioned by codecov the 256 case values are not tested. If you create a test for a fee market transaction then that would be resolved.

Reply:
I'm not aware of any edge cases where a value would need to be less than MAX_INTEGER (since as I understand it, this is always testing for integer overflow and not obscure EIP edge cases) so I don't think this can ever actually happen in reality but I'll add a test case just to be sure.